### PR TITLE
Projects state normalization

### PR DESCRIPTION
**Pull request by kkateq**
1. Normalized state for projects(now repositories).
2. Replaced 'project' with 'repository' (to be consistent with ONO). Some UI components still have to be renamed(minor).
3. Fixed issue with nested repositories for more than the second level.
4. Replaced 'Parent Group' item with breadcrumb.
Along with this task, removed ReduxBreadcrumb and added `Breadcrumb` component for `containers/File`, `containers/Files`
5. Removed  `breadcrumb` ducks code.
6. Fixed flow issues.
7. Unit tests for new reducers, sagas, routes helpers.

**Notes:**

- intentionally removed flow for unit tests files, it produces an enormous amount of errors, seems not relevant for unit tests when it validates the state properties
 
- disabled flow for some UI components, e.g. PullRequestList -> still missing build, for List - issue in flow for the components that use `props.children`
 
- did some removals of using the same flow types for UI (causing weird errors, but probably can be fixed)

### CODE REVIEW

File: src/components/GroupList/GroupList.js [hunk 1]
```diff
@@ -3,10 +3,9 @@
 import React from 'react'
 import { List } from 'material-ui/List'
 import GroupItem from './GroupItem'
-import type { GroupType } from 'ducks/projects'
 
 export type Props = {
-  groups: Array<GroupType>,
+  groups: Array<Object>,
```
Comment (skovhus) [inline src/components/GroupList/GroupList.js]: Why not use the type from ducks anymore?

File: src/components/List/index.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
-/* @flow */
+// TODO: enable flow, issue for components that use props.children
+// https://github.com/facebook/flow/issues/1820
```
Comment (skovhus) [inline src/components/List/index.js]: Children is used other places, shouldn't be any. Try searching for it in some of our other components.

File: src/components/ProjectList/ProjectItem.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ import Col from 'react-bootstrap/lib/Col'
 import Row from 'react-bootstrap/lib/Row'
```
Comment (skovhus) [inline src/components/ProjectList/ProjectItem.js]: Let us rename all these files in this PR when you are touching this anyway.

File: src/components/PullRequestList/PullRequestListItem/index.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-/* @flow */
+// TODO: fix flow for build property
```
Comment (skovhus) [inline src/components/PullRequestList/PullRequestListItem/index.js]: Is it a lot of work to model a build? This really seems like a step in the wrong direction...

File: src/ducks/pullrequests/index.js [hunk 5]
```diff
@@ -37,10 +37,18 @@ export type PullRequestDictionary = {
...
+export type PaginationType = {
+  total: number,
+  pages: Object,
+  pageSize: number,
+  currentPage: number,
+}
+
 export type PullRequestsStateType = {
-  error: ?string,
+  error: Object,
```
Comment (skovhus) [inline src/ducks/pullrequests/index.js]: Please type it

File: src/services/ono/queries/repositories/index.js [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+  description: ?string,
+}
+
+export type ReturnType = {
+  repositories: Array<RepositoryType>,
+  groups: Array<GroupType>,
+}
+
+export function parseRepositories(data: any): ReturnType {
+  const root = (data.data.group || data.data)
```
Comment (skovhus) [inline src/services/ono/queries/repositories/index.js]: Why can root come from multiple places?

File: src/components/List/index.js (hunk 2 above)
Comment (skovhus) [inline src/components/List/index.js]: Try `children?: number | string | React.Element<*> | Array<any>`

File: src/components/GroupList/GroupList.js (hunk 1 above)
Comment (kkateq) [inline src/components/GroupList/GroupList.js]: So not all the types properties are used in UI components, as expected, but flow requires values for them to be specified. So here we are messing up types that we are fetching from GraphQL and types that we are rendering (can be fragments of state) which causes lots of flow issues.

File: src/components/ProjectList/ProjectItem.js (hunk 3 above)
Comment (kkateq) [inline src/components/ProjectList/ProjectItem.js]: Sure, I expected you are going to rename them soon :) but they are still there ...

Comment (skovhus) [inline src/components/ProjectList/ProjectItem.js]: Ok, I can do it then ; )

File: src/components/PullRequestList/PullRequestListItem/index.js (hunk 4 above)
Comment (kkateq) [inline src/components/PullRequestList/PullRequestListItem/index.js]: We can model, but I don't think it makes sense just for the sake of enabling flow here. Component if still not completed and depends on builds results API.

**Comment by skovhus** [comment]
"Squash and merge" 👍

File: src/components/ProjectList/ProjectItem.js (hunk 3 above)
Comment (skovhus) [inline src/components/ProjectList/ProjectItem.js]: So nothing should be called `Project` any more right?

**Pull request merged**